pkg/context/project.go

```diff
@@ -0,0 +1,93 @@
+package context
+
+import (
+	"errors"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+	"io/ioutil"
+	"os"
+	"path"
+	"strmprivacy/strm/pkg/common"
+	"strmprivacy/strm/pkg/entity/project"
+)
+
+const defaultProjectFilename = "default_project"
+
+// ResolveProject resolves the project to use and makes its ID globally available.
+// The value passed through the flag takes precedence, then the value stored in the config dir, and finally
+// a fallback to default project.
+func ResolveProject(f *pflag.FlagSet) {
+
+	activeProjectFilepath := path.Join(common.ConfigPath, defaultProjectFilename)
+	projectFlagValue, _ := f.GetString(ProjectFlag)
+
+	if _, err := os.Stat(activeProjectFilepath); (os.IsNotExist(err) || GetDefaultProject() == "") && projectFlagValue == "" {
+		initDefaultProject()
+	}
+
+	if projectFlagValue != "" {
+		activeProject := project.GetProject(projectFlagValue)
+		if activeProject == nil {
+			message := fmt.Sprintf("Specified project '%v' does not exist, or you do not have access to it.", projectFlagValue)
+			common.CliExit(errors.New(message))
+		}
+		common.ProjectId = activeProject.Id
+	} else {
+		activeProject := project.GetProject(GetDefaultProject())
+		if activeProject == nil {
+			message := fmt.Sprintf("Default project '%v' does not exist, or you do not have access to it. " +
+				"A new default we be set.", GetDefaultProject())
+			fmt.Println(message)
+			initDefaultProject()
+			os.Exit(1)
+		}
+		common.ProjectId = activeProject.Id
+	}
+}
+
+func SetDefaultProject(projectName string) {
+	if project.GetProject(projectName) != nil {
+		saveDefaultProject(projectName)
+		message := "Default project set to: " + projectName
+		log.Infoln(message)
+		fmt.Println(message)
+	} else {
+		message := fmt.Sprintf("No project '%v' found, or you do not have access to it.", projectName)
+		log.Warnln(message)
+		common.CliExit(errors.New(message))
+	}
+}
+
+func GetDefaultProject() string {
+	activeProjectFilepath := path.Join(common.ConfigPath, defaultProjectFilename)
+
+	bytes, err := ioutil.ReadFile(activeProjectFilepath)
+	common.CliExit(err)
+	activeProject := string(bytes)
+	log.Infoln("Current default project is: " + activeProject)
```

Comment:
What's the difference between default and active?

Reply:
Default is the one that is stored locally and used whenever the `--project` flag is not set. Active is whichever actually is active during command execution, and may come from the `--project` flag. I named the default project 'active project' initially, will do another pass on the naming for consistency. Unless you think there is a better way of naming them?

Reply:
I was a bit confused by `default`, because that feels more like a fallback. Active is exactly what it is, so I'd propose to use active.

Reply:
Ah I didn't see your comment before my latest commit 😅  I kept `default` and changed the last remaining usages of `active` (which were very localized) to `resolved`. @bobvandenhoogen is taking this PR over from me on Monday, so perhaps you guys can decide. At this point, I think it's fine to go back to `active`.